libnetwork/datastore/datastore.go

```diff
@@ -149,9 +149,14 @@ func (cfg *ScopeCfg) IsValid() bool {
 
 // Key provides convenient method to create a Key
 func Key(key ...string) string {
-	keychain := append(rootChain, key...)
-	str := strings.Join(keychain, "/")
-	return str + "/"
+	var b strings.Builder
+	for _, parts := range [][]string{rootChain, key} {
+		for _, part := range parts {
+			b.WriteString(part)
+			b.WriteString("/")
```

Comment:
😂 I was about to say "but `strings.Join()` only adds the slashes in between.. and then noticed the `str + "/"` in the original code 👍